micro-rdk/src/common/data_manager.rs

```diff
@@ -107,13 +114,15 @@ where
             sync_interval,
             min_interval,
             part_id,
+            robot_start_time,
         })
     }
 
     pub fn from_robot_and_config(
         cfg: &ConfigResponse,
         app_config: &AppClientConfig,
         robot: Arc<Mutex<LocalRobot>>,
+        robot_start_time: Instant,
```

Comment:
If `LocalRobot` owns its start time, you won't need this to be passed as an additional parameter and it can just be read out of `LocalRobot`.